docs/usage/ssr-hooks.md

```diff
@@ -0,0 +1,96 @@
+---
+name: SSR Hooks
+menu: Usage
+route: '/usage/ssr-hooks'
+---
+
+# Server Side Renderering
+
+## Setup
+
+To SSR your hook, you must ensure `react-dom >= 16.9.0` is installed in your project and then import
+the server module in your test:
+
+```ts
+import { renderHook } from '@testing-library/react-hooks/server'
+```
+
+## Render Hook
+
+`renderHook` when called returns the same result as documented in the
+[API](/reference/api#renderhook-result) but includes an additional argument, `hydrate`:
+
+```ts
+function hydrate(): void
+```
+
+The `hydrate` function is a light wrapper around
+[`ReactDOM.hydrate`](https://reactjs.org/docs/react-dom.html#hydrate) but no arguments are required
+as the library will pass the element & container for you. Remember, certain effects such as
+`useEffect` will not run server side and `hydrate` must be called before those effects are
+ran.`hydrate`is also necessary before the first `act` or `rerender` call. For more information on
+`hydrate` see the [API documentation](/reference/api#hydrate). There is also an
+[example below](/usage/ssr-hooks#example)
+
+## Example
+
+### Hydration
+
+```js
+import { renderHook, act } from '@testing-library/react-hooks/server'
+
+describe('custom hook tests', () => {
+  function useCounter() {
+    const [count, setCount] = useState(0)
+
+    const increment = useCallback(() => setCount(count + 1), [count])
+    const decrement = useCallback(() => setCount(count - 1), [count])
+
+    return { count, increment, decrement }
+  }
+
+  test('should decrement counter', () => {
+    const { result, hydrate } = renderHook(() => useCounter())
+
+    expect(result.current.count).toBe(0)
+
+    // hydrate is called because we want to interact with the hook
+    hydrate()
+
+    act(() => result.current.decrement())
+
+    expect(result.current.count).toBe(-1)
+  })
+})
+```
```

Comment:
````suggestion
## Hydration

The result of rendering you hook is static are not interactive until it is hydrated into the DOM.
This can be done using the `hydrate` function that is returned from `renderHook`.

Consider the `useCounter` example from the [Basic Hooks section](/usage/basic-hooks):

```js
import { useState, useCallback } from 'react'

export default function useCounter() {
  const [count, setCount] = useState(0)
  const increment = useCallback(() => setCount((x) => x + 1), [])
  return { count, increment }
}
```

If we try to call `increment` immediately after server rendering, nothing happens and the hook is
not interactive:

```js
import { renderHook, act } from '@testing-library/react-hooks/server'
import useCounter from './useCounter'

test('should increment counter', () => {
  const { result } = renderHook(() => useCounter(0))

  act(() => {
    result.current.increment()
  })

  expect(result.current.count).toBe(1) // fails as result.current.count is still 0
})
```

We can make the hook interactive by calling the `hydrate` function that is returned from
`renderHook`:

```js
import { renderHook, act } from '@testing-library/react-hooks/server'
import useCounter from './useCounter'

test('should increment counter', () => {
  const { result, hydrate } = renderHook(() => useCounter(0))

  hydrate()

  act(() => {
    result.current.increment()
  })

  expect(result.current.count).toBe(1) // now it passes
})
```

Anything that causes the hook's state to change will not work until `hydrate` is called. This
includes both the [`rerender`](http://localhost:3000/reference/api#rerender) and
[`unmount`](http://localhost:3000/reference/api#unmount) functionality.
````